### PR TITLE
New EIP Request.

### PR DESCRIPTION
**Pull request by jeyakatsa**
@SamWilsn & @lightclient thank you so much for your previous inputs within the previous [EIP](https://github.com/ethereum/EIPs/pull/4649) I requested to pull into this Repo.

Gathered on both your insights, I've implemented a more focused EIP here and would love to get your insights on this.

Available for any questions as to how this project is proceeding.

Thanks again.

### CODE REVIEW

File: EIPS/eip-xx.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
+---
+eip: xx *number to be determined by editor*
```
Comment (lightclient) [inline EIPS/eip-xx.md:2]:
```suggestion
eip: 4780
```

File: EIPS/eip-xx.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+title: smart contracts in java
```
Comment (lightclient) [inline EIPS/eip-xx.md:3]: Please use proper letter casing.

File: EIPS/eip-xx.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+type: standards track
+category: erc
+created: 2022-02-04
+---
+
+## Abstract
+Converting Solidity Keywords into Java Dependencies in order for a library of Smart-Contract implementations to be built in the Java programming language.
+
+## Motivation
+Currently, there are ***200,000 Solidity/Ethereum Developers (Worldwide*** *(source: [TrustNodes](https://www.trustnodes.com/2018/07/22/ethereums-ecosystem-estimated-200000-developers-truffle-seeing-80000-downloads-month))* ***)*** and ***7.1million Java Developers (Worldwide*** *(source: [Daxx](https://www.daxx.com/blog/development-trends/number-software-developers-world#:~:text=According%20to%20SlashData%2C%20the%20number,%2C6%20million%20(source)))* ***)*** respectfully.
```
Comment (lightclient) [inline EIPS/eip-xx.md:17]: Please remove external links.

File: EIPS/eip-xx.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+What if all those Java developers (currently in the millions), could be onboarded into the Ethereum Ecosystem?
```
Comment (lightclient) [inline EIPS/eip-xx.md:19]: Usually we avoid rhetorical questions in EIP.

File: EIPS/eip-xx.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+![](https://github.com/jeyakatsa/EIPs/blob/master/assets/eip-xx/Java-Abstraction-Visualization-display.jpg)
+
+## Specification
```
Comment (lightclient) [inline EIPS/eip-xx.md:23]: An EIP should describe how the proposal should work. I don't view this specification as sufficient to converting a solidity program into java code or vice versa.

File: EIPS/eip-xx.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+            e.printStackTrace();
+        }
+
+        decryptedTextBytes.toString();
+    }
+}
+```
+
+## Backwards Compatibility
+These Smart-Contract Java abstractions are meant to be compatible with the overall Ethereum Execution and Consensus layers respectfully.
```
Comment (lightclient) [inline EIPS/eip-xx.md:103]: I'm not sure how this would be compatible with the consensus layer given it has no execution?

File: EIPS/eip-xx.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+
+## Security Considerations
+A 51% attack of these contracts seems to be the biggest security implication concerning these new abstractions. But as stated once by Vitalik, it seems implausible for a validator to get away with a 51% attack of any token (or smart-contract) running on Ethereum safely.
```
Comment (lightclient) [inline EIPS/eip-xx.md:106]: Not sure what you mean by 51% attack here? Best you can do is censor, or trick light clients into accepting irregular state transitions.

File: EIPS/eip-xx.md [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+
+The second security concern is the expanding base of smart-contracts from Solidity into Java. Solidity is currently not as widely known by developers as Java thus making it a "blessing in disguise" in regards to how secure the contracts running on Ethereum are. Expanding the creation of these smart-contracts from Solidity into Java will blossom the development of the ecosystem, while also attracting maliciousness (hence the larger an ecosystem grows, the more "criminals" it attracts).
+
+#### References
```
Comment (lightclient) [inline EIPS/eip-xx.md:110]: This section should be removed.

**Review by lightclient — CHANGES_REQUESTED** [review]

File: EIPS/eip-xx.md (hunk 5 above)
Comment (jeyakatsa) [reply EIPS/eip-xx.md:23]: Understood.

File: EIPS/eip-xx.md (hunk 6 above)
Comment (jeyakatsa) [reply EIPS/eip-xx.md:103]: Still working on the execution, will return once that's implemented. Thank you @lightclient

File: EIPS/eip-xx.md (hunk 7 above)
Comment (jeyakatsa) [reply EIPS/eip-xx.md:106]: Will rework this and return with a more focused consideration.

File: EIPS/eip-xx.md (hunk 8 above)
Comment (jeyakatsa) [reply EIPS/eip-xx.md:110]: Noted.

File: EIPS/eip-xx.md (hunk 2 above)
Comment (jeyakatsa) [reply EIPS/eip-xx.md:3]: Done.

File: EIPS/eip-xx.md (hunk 3 above)
Comment (jeyakatsa) [reply EIPS/eip-xx.md:17]: Done

File: EIPS/eip-4780.md [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+
+## Security Considerations
+*This section is still being reworked*
+
+#### References
+- [Java Smart Contract Abstraction for Ethereum R&D](https://github.com/jeyakatsa/ethereum-smart-contract-java-abstraction/tree/main/R%26D-files)
+- [Java Smart Contract Abstraction for Ethereum](https://github.com/jeyakatsa/Ethereum-Smart-Contract-Java-Abstraction)
+- [New-ERC Token Proposal](https://ethresear.ch/t/a-new-erc-token-proposal/11540)
+- [Light-Client Token Creation Proposal](https://ethresear.ch/t/light-client-custom-token-creation-proposal/11433)
+
```
Comment (MicahZoltu) [inline EIPS/eip-4780.md:111]:
```suggestion
```
External links are not appropriate for an EIP.  EIPs should be pure technical specifications/standards and not depend on any external references.

**Comment by MicahZoltu** [comment]
I don't think this is appropriate for an EIP/standard.  One can write a Java => EVM compiler without authoring any formal specification/standard, and if you did want to formalize your implementation you could do so in your own documentation without submitting it as an EIP.  This is similar to how Solidity doesn't have any EIP associated with it, there just is a team that has written the compiler and they iterate on and document it independent from the EIP process.

**Comment by jeyakatsa** [comment]
> I don't think this is appropriate for an EIP/standard. One can write a Java => EVM compiler without authoring any formal specification/standard, and if you did want to formalize your implementation you could do so in your own documentation without submitting it as an EIP. This is similar to how Solidity doesn't have any EIP associated with it, there just is a team that has written the compiler and they iterate on and document it independent from the EIP process.

Noted.

**Pull request closed without merging**